core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -401,31 +487,71 @@ func (o *OptimismL1Oracle) getEcotoneGasPrice(ctx context.Context) (*big.Int, er
+	}
+	if rpcBatchCalls[3].Error != nil {
+		return nil, fmt.Errorf("%s call failed in a batch: %w", blobBaseFeeScalarMethod, err)
+	}
+	if rpcBatchCalls[4].Error != nil {
+		return nil, fmt.Errorf("%s call failed in a batch: %w", decimalsMethod, err)
 	}
 
-	l1GasUsedResult := *(rpcBatchCalls[0].Result.(*string))
-	l1FeeResult := *(rpcBatchCalls[1].Result.(*string))
+	// Extract values from responses
+	l1BaseFeeResult := *(rpcBatchCalls[0].Result.(*string))
+	baseFeeScalarResult := *(rpcBatchCalls[1].Result.(*string))
+	blobBaseFeeResult := *(rpcBatchCalls[2].Result.(*string))
+	blobBaseFeeScalarResult := *(rpcBatchCalls[3].Result.(*string))
+	decimalsResult := *(rpcBatchCalls[4].Result.(*string))
 
-	l1GasUsedBytes, err := hexutil.Decode(l1GasUsedResult)
+	// Decode the responses into bytes
+	l1BaseFeeBytes, err := hexutil.Decode(l1BaseFeeResult)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode %s rpc result: %w", OPStackGasOracle_getL1GasUsed, err)
+		return nil, fmt.Errorf("failed to decode %s rpc result: %w", l1BaseFeeMethod, err)
 	}
-	l1FeeBytes, err := hexutil.Decode(l1FeeResult)
+	baseFeeScalarBytes, err := hexutil.Decode(baseFeeScalarResult)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode %s rpc result: %w", OPStackGasOracle_getL1Fee, err)
+		return nil, fmt.Errorf("failed to decode %s rpc result: %w", baseFeeScalarMethod, err)
 	}
+	blobBaseFeeBytes, err := hexutil.Decode(blobBaseFeeResult)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode %s rpc result: %w", blobBaseFeeMethod, err)
+	}
+	blobBaseFeeScalarBytes, err := hexutil.Decode(blobBaseFeeScalarResult)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode %s rpc result: %w", blobBaseFeeScalarMethod, err)
+	}
+	decimalsBytes, err := hexutil.Decode(decimalsResult)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode %s rpc result: %w", decimalsMethod, err)
+	}
+
+	// Convert bytes to big int for calculations
+	l1BaseFee := new(big.Int).SetBytes(l1BaseFeeBytes)
+	baseFeeScalar := new(big.Int).SetBytes(baseFeeScalarBytes)
+	blobBaseFee := new(big.Int).SetBytes(blobBaseFeeBytes)
+	blobBaseFeeScalar := new(big.Int).SetBytes(blobBaseFeeScalarBytes)
+	decimals := new(big.Int).SetBytes(decimalsBytes)
+
+	o.logger.Debugw("gas price parameters", "l1BaseFee", l1BaseFee, "baseFeeScalar", baseFeeScalar, "blobBaseFee", blobBaseFee, "blobBaseFeeScalar", blobBaseFeeScalar, "decimals", decimals)
+
+	// Scaled gas price = baseFee * 16 * baseFeeScalar + blobBaseFee * blobBaseFeeScalar
+	scaledBaseFee := new(big.Int).Mul(l1BaseFee, baseFeeScalar)
+	scaledBaseFee = new(big.Int).Mul(scaledBaseFee, big.NewInt(16))
+	scaledBlobBaseFee := new(big.Int).Mul(blobBaseFee, blobBaseFeeScalar)
+	scaledGasPrice := new(big.Int).Add(scaledBaseFee, scaledBlobBaseFee)
 
-	l1GasUsed := new(big.Int).SetBytes(l1GasUsedBytes)
-	l1Fee := new(big.Int).SetBytes(l1FeeBytes)
+	// Gas price = scaled gas price / (16 * 10 ^ decimals)
```

Comment:
nit: Can you add links for both Ecotone and Fjord that will verify this formula?

Reply:
I've added the links and some context in the latest commit